presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1732,6 +1734,36 @@ public void dropView(ConnectorSession session, SchemaTableName viewName)
         return tableNames.build();
     }
 
+    @Override
+    public Map<String, Object> getSchemaProperties(ConnectorSession session, CatalogSchemaName schemaName)
+    {
+        if (!filterSchema(schemaName.getSchemaName())) {
+            return Collections.emptyMap();
+        }
+        String warehouse = metastore.getConfigValue(WAREHOUSE.getVarname()).orElseGet(() -> WAREHOUSE.getDefaultVal().toString());
+        Optional<Database> db = metastore.getDatabase(schemaName.getSchemaName());
+        if (db.isPresent()) {
+            Optional<String> location = db.get().getLocation();
+            // this is currently the only property for Hive
+            // (only shown when it's not the default location)
+            if (location.isPresent() && !location.get().startsWith(warehouse)) {
```

Comment:
I think this condition is not enough. Is it possible that one could create external table under `warehouse` location? Like pointing to data of another managed table? I do think there is a check for that in Hive and this allowed. We need to verify full path, not only prefix.

Please add test for that.

Reply:
Hmm... I was debating this as well. I'd have to hardcode a convention then, name that the default name for any database is `<datewarehouse.dir>/<databasename>.db`
It's not clear what it means if somebody now manually creates a directory inside the `datawarehouse.dir` and then places a schema (not table, there are no external schemas) in there.
I can certain verify that the path is *exactly* `<datewarehouse.dir>/<databasename>.db` and show it otherwise.

Reply:
The default is `<warehouse.dir>/<schemaname>.db`, but any path is allowed (without ending in .db). So I think we should verify `<warehouse.dir>/<schemaname>` exactly.

But people can vary the "extension" too. The file based hive metastore has no extensions.
So if you name your location `<warehouse.dir>/<schemaname>.mydb` should I show the location or not?

Also looks like there are no further checks. I can create a schema in `/a/b/c` and also in `/a/b`, then create a table `t` in `/a/b`. Now we have a schema `c` and table `t` in the same directory.

@kokosing

Reply:
BTW. Hive always shows the location whether it's default or not.
Perhaps it's better to follow that?